core/pkg/sync/builder/syncbuilder.go

```diff
@@ -24,6 +23,8 @@ import (
 
 const (
 	syncProviderFile       = "file"
+	syncProviderFsNotify   = "fsnotify"
+	syncProviderFileInfo   = "fileinfo"
```

Comment:
This commit implements, what _I think_ we want for configs.  The user can express a global sync-source config with one of three values

- "file" is the existing default, likely widely used in the wild. This will trigger a switch/case on the value of the `KUBERNETES_SERVICE_HOST` Env-var.  If this var has a value, we return an fsnotify-driven sync, otherwise we return a os.stat/fileinfo-driven sync.
- "fsnotify" enables the user to explicitly specify an fsnotify-driven sync
- "fileinfo" enables the user to explicitly specify an os.Stat/fs.FileInfo-driven sync

Reply:
Could you please update the sync configuration documentation with this change?

https://github.com/open-feature/flagd/blob/main/docs/reference/sync-configuration.md

Reply:
I'm in favor of this default "file" behavior.